addons/l10n_in/models/account_invoice.py

```diff
@@ -97,3 +97,40 @@ def _l10n_in_get_warehouse_address(self):
         # TO OVERRIDE
         self.ensure_one()
         return False
+
+    def _l10n_in_get_hsn_summary(self):
+        for move in self:
+            hsn_summary = {}
+
+            def grouping_key_generator(base_line, tax_values):
+                hsn = base_line['product'].l10n_in_hsn_code
+                gst_tag_ids = self.env["account.account.tag"]
+                line_code = ""
+                tax_line_tag_ids = tax_values['tag_ids']
+                for gst in ["cgst", "sgst", "igst"]:
+                    gst_tag = self.env.ref("l10n_in.tax_tag_%s" % (gst))
+                    gst_tag_ids += gst_tag
+                    if gst_tag.id in tax_line_tag_ids:
+                        line_code = gst
+                cess_tag_ids = self.env.ref("l10n_in.tax_tag_cess") + self.env.ref("l10n_in.tax_tag_state_cess")
+                if any(tag.id in tax_line_tag_ids for tag in cess_tag_ids):
+                    line_code = "cess"
+                tax_ids = base_line['record'].tax_ids.flatten_taxes_hierarchy()
```

Comment:
This flatten_taxes_hierarchy is a bit cumbersome, but might be necessary.  I see that for the GSTR report, we have a different query for it, so we can deduce the gst rate more easily.